### DISCUSSION TITLE
Detecting Mic Connection

### DISCUSSION THREAD

**Discussion by corfoto4** [Sapd/HeadsetControl#328] [General]
Hey! I'm looking for a way to detect when my arctis nova 7 connects to the usb dongle on my pc. I have looked through this repository and have found that it can get the status of the headset.

Is there a way to detect on interrupt when the headset connects and disconnects? Maybe I can use the basis of this repository and make my own program? Then I can set up my own programs/scripts.

Thank You :)

**Comment by Sapd** [comment]
You probably could take advantage of the --connected parameter and poll (which however is not 100% certain, it checks battery level to see if something is connected).

Probably something like this (without testing it)

```bash
while true; do
    if ./headsetcontrol --connected; then
        notify-send "Headset Connected" "Your headset is now connected."
        # Optional: break 
        # break
    fi
    # Wait for a bit before checking again
    sleep 5
done
```

**Reply by corfoto4** [reply]
How often should I poll? like once a minute?

Otherwise, do you think there is a way to convert it to interrupt? I'm going by the steelseries software engine. Surely that uses interrupt (which I might be able to detect) or if not then would that mean it polls?

I basically want to bypass the steelseries software (sonarr) and have it so that when my headset connects, it auto switches over the output and input devices (and switch back on disconnect)

Similar to my old corsair headset

**Reply by Sapd** [reply]
You need to experiment a bit with that. You could poll it every second in theory.

> Otherwise, do you think there is a way to convert it to interrupt? I'm going by the steelseries software engine. Surely that uses interrupt (which I might be able to detect) or if not then would that mean it polls?

It probably does something what hid_read does, which is a kind of interrupt yes.
You could experiment a bit with
```
headsetcontrol --dev -- --device 0x1b1c:0x1b27 --receive
``` 
(there is however a small bug where the timeout parameter does not work as expected, it should actually be blocking when no timeout is provided, but it will show no data to read. Will be fixed in next version, but you could just open up `dev.c` and modify `int timeout` and set it to -1 )

It will probably then give you some data when the headset is connected or disconnected.

But in the end there is probably no big difference to the battery command and it might be even required (often the battery command is just a status command)
